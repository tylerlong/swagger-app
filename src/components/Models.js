import React from 'react'
import R from 'ramda'
import { Button } from 'antd'
import { connect } from 'react-redux'

import { addModel } from '../actions'

class Models extends React.Component {
  render () {
    return (
      <div>
        <h2>Models</h2>
        <div className='primary-button'>
          <Button type='primary' size='large' onClick={this.props.addModel}>Add model</Button>
        </div>
      </div>
    )
  }
}

export default connect(R.identity, { addModel })(Models)
