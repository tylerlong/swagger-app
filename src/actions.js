import axios from 'axios'
import { message } from 'antd'

export const setProp = (path, value) => ({ type: 'SET_PROP', path, value })

export const loadState = () => async dispatch => {
  const res = await axios.get('./sample.json')
  dispatch({ type: 'SET_STATE', state: res.data })
  message.success(`Data loaded`)
}

export const addPermission = () => (dispatch, getState) => {
  dispatch({ type: 'ADD_PERMISSION', name: 'Permission name', description: 'Permission description' })
  dispatch({ type: 'SET_PROP', path: ['metadata', 'activePermissionIndex'], value: getState().permissions.length - 1 }) // make last Collapse panel active
  message.success(`Permission added`)
}

export const deletePermission = () => (dispatch, getState) => {
  dispatch({ type: 'DELETE_PERMISSION' })
  const state = getState()
  if (state.metadata.activePermissionIndex === state.permissions.length) {
    dispatch({ type: 'SET_PROP', path: ['metadata', 'activePermissionIndex'], value: state.permissions.length - 1 }) // make last Collapse panel active
  }
  message.success(`Permission deleted`)
}

export const addPathParameter = () => (dispatch, getState) => {
  dispatch({ type: 'ADD_PATH_PARAMETER', name: 'Name', description: 'Description', enum: [] })
  dispatch({ type: 'SET_PROP', path: ['metadata', 'activePathParameterIndex'], value: getState().pathParameters.length - 1 }) // make last Collapse panel active
  message.success(`Path parameter added`)
}

export const deletePathParameter = () => (dispatch, getState) => {
  dispatch({ type: 'DELETE_PATH_PARAMETER' })
  const state = getState()
  if (state.metadata.activePathParameterIndex === state.pathParameters.length) {
    dispatch({ type: 'SET_PROP', path: ['metadata', 'activePathParameterIndex'], value: state.pathParameters.length - 1 }) // make last Collapse panel active
  }
  message.success(`Path parameter deleted`)
}

export const addPath = () => dispatch => {
  dispatch({ type: 'ADD_PATH' })
  message.success(`Path added`)
}

export const addModel = () => dispatch => {
  dispatch({ type: 'ADD_MODEL' })
  message.success(`Model added`)
}

export const movePermissionUp = () => dispatch => {
  dispatch({ type: 'MOVE_PERMISSION_UP' })
  message.success(`Permission moved up`)
}

export const movePermissionDown = () => dispatch => {
  dispatch({ type: 'MOVE_PERMISSION_DOWN' })
  message.success(`Permission moved down`)
}

export const movePathParameterUp = () => dispatch => {
  dispatch({ type: 'MOVE_PATH_PARAMETER_UP' })
  message.success(`PathParameter moved up`)
}

export const movePathParameterDown = () => dispatch => {
  dispatch({ type: 'MOVE_PATH_PARAMETER_DOWN' })
  message.success(`PathParameter moved down`)
}
