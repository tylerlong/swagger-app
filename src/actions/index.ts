export * from './app'
export * from './permissions'
export * from './pathParameters'
export * from './paths'
export * from './models'
