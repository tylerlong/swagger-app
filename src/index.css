.ant-tabs {
  margin-top: 16px;
  border-top: 1px solid #ddd;
  padding-top: 16px;
}

.permission-form-item,
.path-parameter-form-item {
  border-bottom: 1px dotted #ddd;
  margin-bottom: 16px;
}

.primary-button {
  text-align: center;
  margin-bottom: 16px;
  padding-bottom: 16px;
  border-bottom: 1px dotted #ddd;
}

.delete-button {
  text-align: center;
  margin-bottom: 16px;
}
