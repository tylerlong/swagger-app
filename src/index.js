import './index.css'

import React from 'react'
import ReactDOM from 'react-dom'
import { message, Input, Form } from 'antd'
import axios from 'axios'

class App extends React.Component {
  constructor (props) {
    super(props)
    this.state = {
      name: '',
      version: ''
    }

    this.handleChangeName = this.handleChangeName.bind(this)
    this.handleChangeVersion = this.handleChangeVersion.bind(this)
  }
  componentDidMount () {
    axios.get('./sample.json').then(res => {
      this.setState(res.data)
      message.success('state loaded')
    })
  }
  handleChangeName (event) {
    this.setState({ name: event.target.value })
  }
  handleChangeVersion (event) {
    this.setState({ version: event.target.value })
  }
  render () {
    const formItemLayout = {
      labelCol: {
        xs: { span: 24 },
        sm: { span: 6 }
      },
      wrapperCol: {
        xs: { span: 24 },
        sm: { span: 12 }
      }
    }
    return (
      <div>
        <h1>{this.state.name} {this.state.version}</h1>
        <Form>
          <Form.Item {...formItemLayout} label='Name'>
            <Input placeholder='Name' size='large' value={this.state.name} onChange={this.handleChangeName} />
          </Form.Item>
          <Form.Item {...formItemLayout} label='Version'>
            <Input placeholder='Version' size='large' value={this.state.version} onChange={this.handleChangeVersion} />
          </Form.Item>
        </Form>
      </div>
    )
  }
}

ReactDOM.render(<App />, document.getElementById('root'))
