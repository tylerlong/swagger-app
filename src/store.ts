import { createStore, applyMiddleware } from 'redux'
import { createLogicMiddleware } from 'redux-logic'

import reducer from './reducers'
import logics from './logics'

const logicMiddleware = createLogicMiddleware(logics)
const store = createStore(
  reducer,
  applyMiddleware(logicMiddleware)
)

export default store
