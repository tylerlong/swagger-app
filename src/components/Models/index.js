import React from 'react'
import R from 'ramda'
import { Button, Collapse } from 'antd'
import { connect } from 'react-redux'

import FormItem from './FormItem'
import { addModel, setProp } from '../../actions'

class Models extends React.Component {
  render () {
    const { models, addModel } = this.props
    return (
      <div>
        <h2>Models</h2>
        <Collapse accordion>
          {R.sort(R.comparator((a, b) => R.toLower(a.name) < R.toLower(b.name)), models).map(model => {
            return (
              <Collapse.Panel header={model.name} key={model.createdAt}>
                <FormItem index={R.findIndex(R.propEq('createdAt', model.createdAt), models)} />
              </Collapse.Panel>
            )
          })}
        </Collapse>
        <div className='primary-button'>
          <Button type='primary' size='large' onClick={addModel}>Add model</Button>
        </div>
      </div>
    )
  }
}

export default connect(R.identity, { addModel, setProp })(Models)
