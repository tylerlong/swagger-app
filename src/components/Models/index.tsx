import React from 'react'
import R from 'ramda'
import { Button, Collapse, Icon } from 'antd'
import { connect } from 'react-redux'

import FormItem from './FormItem'
import { addModel } from '../../actions'
import { orderBy } from '../../utils'

class Models extends React.Component {
  render () {
    console.log(`render Models`)
    const { models, addModel } = this.props
    return (
      <div>
        <h2>Models</h2>
        <Collapse accordion>
          {orderBy(R.prop('name'), models).map(model => {
            return (
              <Collapse.Panel header={model.name} key={model.createdAt}>
                <FormItem index={R.findIndex(R.propEq('createdAt', model.createdAt), models)} />
              </Collapse.Panel>
            )
          })}
        </Collapse>
        <Button type='primary' size='large' onClick={addModel}><Icon type='plus' />Add model</Button>
      </div>
    )
  }
}

export default connect(R.pick(['models']), { addModel })(Models)
