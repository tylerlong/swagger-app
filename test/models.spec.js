/* eslint-env jest */
import React from 'react'
import { Provider } from 'react-redux'
import { mount } from 'enzyme'
import toJson from 'enzyme-to-json'

import store from './store'
import { setState, addModel, deleteModel, addModelProperty, deleteModelProperty } from '../src/actions'
import Models from '../src/components/Models'

const initialState = { models: [{ createdAt: 123, name: '1', properties: [] }, { createdAt: 456, name: '2', properties: [] }] }

beforeEach(() => {
  store.dispatch(setState(initialState))
  store.resetActions()
})

const getWrapper = () => {
  return mount(
    <Provider store={store}>
      <Models />
    </Provider>
  )
}

describe('test model', () => {
  test('add model', () => {
    store.dispatch(addModel())
    const models = store.getState().models
    expect(models.length).toEqual(3)
    expect(models[2].name).toEqual('Name')
  })
  test('delete model', () => {
    store.dispatch(deleteModel(1))
    const models = store.getState().models
    expect(models.length).toEqual(1)
    expect(models[0].name).toEqual('1')
  })
  test('add model property', () => {
    store.dispatch(addModelProperty(1))
    const properties = store.getState().models[1].properties
    expect(properties.length).toEqual(1)
    expect(properties[0].name).toEqual('name')
  })
  test('delete model property', () => {
    store.dispatch(addModelProperty(1))
    store.dispatch(deleteModelProperty(1, 0))
    const properties = store.getState().models[1].properties
    expect(properties.length).toEqual(0)
  })
  test('view model', () => {
    const wrapper = getWrapper()
    expect(wrapper.find('.ant-collapse-item').length).toEqual(2)
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})
