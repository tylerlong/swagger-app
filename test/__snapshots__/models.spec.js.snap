// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test model view model 1`] = `
<Provider
  store={
    Object {
      "actions": Array [],
      "dispatch": [Function],
      "getState": [Function],
      "logicMiddleware": [Function],
      "replaceReducer": [Function],
      "resetActions": [Function],
      "subscribe": [Function],
      "whenComplete": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <Connect(Models)>
    <Models
      addModel={[Function]}
      models={
        Array [
          Object {
            "createdAt": 123,
            "name": "1",
            "properties": Array [],
          },
          Object {
            "createdAt": 456,
            "name": "2",
            "properties": Array [],
          },
        ]
      }
    >
      <div>
        <h2>
          Models
        </h2>
        <Collapse
          accordion={true}
          bordered={true}
          openAnimation={
            Object {
              "appear": [Function],
              "enter": [Function],
              "leave": [Function],
            }
          }
          prefixCls="ant-collapse"
        >
          <Collapse
            accordion={true}
            bordered={true}
            className=""
            destroyInactivePanel={false}
            onChange={[Function]}
            openAnimation={
              Object {
                "appear": [Function],
                "enter": [Function],
                "leave": [Function],
              }
            }
            prefixCls="ant-collapse"
          >
            <div
              className="ant-collapse"
            >
              <CollapsePanel
                destroyInactivePanel={false}
                header="1"
                headerClass=""
                isActive={false}
                onItemClick={[Function]}
                openAnimation={
                  Object {
                    "appear": [Function],
                    "enter": [Function],
                    "leave": [Function],
                  }
                }
                prefixCls="ant-collapse"
                showArrow={true}
              >
                <div
                  className="ant-collapse-item"
                >
                  <div
                    aria-expanded={false}
                    className="ant-collapse-header"
                    onClick={[Function]}
                    role="tab"
                  >
                    <i
                      className="arrow"
                    />
                    1
                  </div>
                  <Animate
                    animation={
                      Object {
                        "appear": [Function],
                        "enter": [Function],
                        "leave": [Function],
                      }
                    }
                    component=""
                    componentProps={Object {}}
                    exclusive={true}
                    onAppear={[Function]}
                    onEnd={[Function]}
                    onEnter={[Function]}
                    onLeave={[Function]}
                    showProp="isActive"
                    transitionAppear={false}
                    transitionEnter={true}
                    transitionLeave={true}
                  >
                    <AnimateChild
                      animation={
                        Object {
                          "appear": [Function],
                          "enter": [Function],
                          "leave": [Function],
                        }
                      }
                      transitionAppear={false}
                      transitionEnter={true}
                      transitionLeave={true}
                    >
                      <PanelContent
                        destroyInactivePanel={false}
                        isActive={false}
                        prefixCls="ant-collapse"
                      />
                    </AnimateChild>
                  </Animate>
                </div>
              </CollapsePanel>
              <CollapsePanel
                destroyInactivePanel={false}
                header="2"
                headerClass=""
                isActive={false}
                onItemClick={[Function]}
                openAnimation={
                  Object {
                    "appear": [Function],
                    "enter": [Function],
                    "leave": [Function],
                  }
                }
                prefixCls="ant-collapse"
                showArrow={true}
              >
                <div
                  className="ant-collapse-item"
                >
                  <div
                    aria-expanded={false}
                    className="ant-collapse-header"
                    onClick={[Function]}
                    role="tab"
                  >
                    <i
                      className="arrow"
                    />
                    2
                  </div>
                  <Animate
                    animation={
                      Object {
                        "appear": [Function],
                        "enter": [Function],
                        "leave": [Function],
                      }
                    }
                    component=""
                    componentProps={Object {}}
                    exclusive={true}
                    onAppear={[Function]}
                    onEnd={[Function]}
                    onEnter={[Function]}
                    onLeave={[Function]}
                    showProp="isActive"
                    transitionAppear={false}
                    transitionEnter={true}
                    transitionLeave={true}
                  >
                    <AnimateChild
                      animation={
                        Object {
                          "appear": [Function],
                          "enter": [Function],
                          "leave": [Function],
                        }
                      }
                      transitionAppear={false}
                      transitionEnter={true}
                      transitionLeave={true}
                    >
                      <PanelContent
                        destroyInactivePanel={false}
                        isActive={false}
                        prefixCls="ant-collapse"
                      />
                    </AnimateChild>
                  </Animate>
                </div>
              </CollapsePanel>
            </div>
          </Collapse>
        </Collapse>
        <Button
          clicked={false}
          ghost={false}
          loading={false}
          onClick={[Function]}
          prefixCls="ant-btn"
          size="large"
          type="primary"
        >
          <button
            className="ant-btn ant-btn-primary ant-btn-lg"
            onClick={[Function]}
            onMouseUp={[Function]}
            type="button"
          >
            <Icon
              type="plus"
            >
              <i
                className="anticon anticon-plus"
              />
            </Icon>
            <span>
              Add model
            </span>
          </button>
        </Button>
      </div>
    </Models>
  </Connect(Models)>
</Provider>
`;
